test/common/buffer/watermark_buffer_test.cc

```diff
@@ -431,22 +431,32 @@ TEST_F(WatermarkBufferTest, OverflowWatermarkDisabledOnVeryHighValue) {
 
   // Make sure the overflow threshold will be above std::numeric_limits<uint32_t>::max()
   Runtime::LoaderSingleton::getExisting()->mergeValues({{"envoy.buffer.overflow_multiplier", "3"}});
-  buffer1.setWatermarks((std::numeric_limits<uint32_t>::max() / 3) + 1);
+  const uint32_t high_watermark_threshold = (std::numeric_limits<uint32_t>::max() / 3) + 1;
+  buffer1.setWatermarks(high_watermark_threshold);
 
   // Add many segments instead of full uint32_t::max to get around std::bad_alloc exception
   const uint32_t segment_denominator = 128;
   const uint32_t big_segment_len = std::numeric_limits<uint32_t>::max() / segment_denominator + 1;
   const std::string big_segment_str = std::string(big_segment_len, 'a');
+  // Fill the buffer with more than uint32_t::max bytes
+  EXPECT_GT(static_cast<uint64_t>(segment_denominator) * big_segment_len,
+            std::numeric_limits<uint32_t>::max());
   for (uint32_t i = 0; i < segment_denominator; ++i) {
     buffer1.add(big_segment_str.data(), big_segment_len);
   }
+  EXPECT_GT(buffer1.length(), std::numeric_limits<uint32_t>::max());
   EXPECT_EQ(1, high_watermark_buffer1);
   EXPECT_EQ(0, overflow_watermark_buffer1);
-  buffer1.add(TEN_BYTES, 10);
+  EXPECT_LT(buffer1.length(), high_watermark_threshold * UINT64_C(3));
```

Comment:
Can you define an uint64 constant for this 3 that appears here and in the high_watermark_threshold computation?

Reply:
Done